pkg/testing/fixture.go

```diff
@@ -1276,6 +1276,45 @@ func FindComponentsDir(dir, version string) (string, error) {
 	return componentsDir, nil
 }
 
+// FindRunDir identifies the directory that holds the run folder.
+func FindRunDir(fixture *Fixture) (string, error) {
+	agentWorkDir := fixture.WorkDir()
+	if pf := fixture.PackageFormat(); pf == "deb" || pf == "rpm" {
+		// these are hardcoded paths in packages.yml
+		agentWorkDir = "/var/lib/elastic-agent"
+	}
+
+	version := fixture.Version()
+
+	dataDir := filepath.Join(agentWorkDir, "data")
+	agentVersions, err := os.ReadDir(dataDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to read contents of the data directory %s: %w", dataDir, err)
+	}
+	var versionDir string
+	for _, fi := range agentVersions {
+		filename := fi.Name()
+		if strings.HasPrefix(filename, "elastic-agent-") && fi.IsDir() {
+			if version != "" && filename[:len(filename)-7] != "elastic-agent-"+version {
```

Comment:
Nit: add a comment explaining the choice of `7` here.  I know it's for the hash but future me reading this code will probably not immediately realize that 🙂.

Reply:
thanks for catching that I did realise that it will improve more the code readability if I deduplicated the same code between `FindComponentsDir` and `FindRunDir`. I did add also a comment about the hash here 0318eabbaac05abd71cf4be3a9c795d93d8047b3. Feel free to unresolve if you spot anything else 🙂